analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/IsType.cs

```diff
@@ -28,7 +28,7 @@ internal sealed class IsType : BranchingProcessor<IIsTypeOperationWrapper>
         IIsTypeOperationWrapper.FromOperation(operation);
 
     protected override SymbolicConstraint BoolConstraintFromOperation(ProgramState state, IIsTypeOperationWrapper operation) =>
-        null;
+        state[operation.ValueOperand]?.HasConstraint(ObjectConstraint.Null) is true ? BoolConstraint.False : null;
```

Comment:
We could also learn true if ValueOperand has the "NotNullConstraint" and the type of the value is implicitly convertible to the type tested against. I think we have something similar in pattern handling.

Reply:
Yes, we could. I wouldn't go out of scope here. I don't see too many use cases for checking something that could be implicitly convertible.